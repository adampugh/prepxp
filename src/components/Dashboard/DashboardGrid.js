import React, { Component } from "react";

import DashboardSidebar from "./DashboardSidebar";
import DashboardList from "./dashboardList";

class DashboardGrid extends Component {
    render(props) {
        return (
            <section className="container">
                <div className="dashboard__grid">
                    <DashboardSidebar />
                    <div className="dashboard__grid__lists">
                        <h1>Hello, {this.props.name}!</h1>
                        {this.props.lists.length > 0 ? this.props.lists.map((list, index) => <DashboardList key={`${list.title + index}`} list={list} />)
                            : 
                            <div className="dashboard__empty">
                                <h2>You don't have any lists yet, add or create some!</h2>
                                <button className="btn">Lists</button>
                            </div> 
                        }
                    </div>
                </div>
            </section>
        );
    }
}

export default DashboardGrid;