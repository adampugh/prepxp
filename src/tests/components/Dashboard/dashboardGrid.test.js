import React from "react";
import { shallow } from "enzyme";

import DashboardGrid from "../../../components/Dashboard/DashboardGrid";
import user from "../../fixtures/userLists";

describe("<DashboardGrid />", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(<DashboardGrid lists={user.lists}  />);
        expect(wrapper).toMatchSnapshot();
    });
});