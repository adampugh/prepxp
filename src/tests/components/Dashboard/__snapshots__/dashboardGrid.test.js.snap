// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardGrid /> should match snapshot 1`] = `
<div
  className="dashboard__wrapper"
>
  <section
    className="container"
  >
    <div
      className="dashboard__grid"
    >
      <dashboardSidebar />
      <div
        className="dashboard__grid__lists"
      >
        <h1>
          Hello, 
          !
        </h1>
        <dashboardList
          key="JavaScript Developer0"
          list={
            Object {
              "questions": Array [
                Object {
                  "answer": "JavaScript is a language for the web.",
                  "question": "What is the difference between let and var?",
                },
                Object {
                  "answer": "Brendan Eich",
                  "question": "Who created JavaScript?",
                },
              ],
              "title": "JavaScript Developer",
            }
          }
        />
        <dashboardList
          key="PHP Developer1"
          list={
            Object {
              "questions": Array [
                Object {
                  "answer": "No-one really knows.",
                  "question": "What is PHP exactly?",
                },
              ],
              "title": "PHP Developer",
            }
          }
        />
      </div>
    </div>
  </section>
</div>
`;
